diff --git a/pysc2/bin/agent.py b/pysc2/bin/agent.py
--- a/pysc2/bin/agent.py
+++ b/pysc2/bin/agent.py
@@ -52,6 +52,7 @@ flags.DEFINE_integer("max_agent_steps", 0, "Total agent steps.")
 flags.DEFINE_integer("game_steps_per_episode", None, "Game steps per episode.")
 flags.DEFINE_integer("max_episodes", 1, "Total episodes.")
 flags.DEFINE_integer("step_mul", 1, "Game steps per agent step.")
+flags.DEFINE_integer("random_seed", None, "Random seed used to initialize the game.")
 
 flags.DEFINE_string("agent", "pysc2.agents.random_agent.RandomAgent",
                     "Which agent to run, as a python path to an Agent class.")
@@ -100,6 +101,7 @@ def run_thread(agent_classes, players, map_name, visualize):
       step_mul=FLAGS.step_mul,
       game_steps_per_episode=FLAGS.game_steps_per_episode,
       disable_fog=FLAGS.disable_fog,
+      random_seed=FLAGS.random_seed,
       visualize=visualize) as env:
     env = available_actions_printer.AvailableActionsPrinter(env)
     agents = [agent_cls() for agent_cls in agent_classes]
